### DISCUSSION TITLE
%pagecount% and background pages

### DISCUSSION THREAD

**Discussion by jwilliams-shadowsoft** [jgraph/drawio#3378] [Q&A]
I love the fact that drawio now has background pages. I'm an engineer, and when I make technical drawings for my customers the pagination is always of the form:

```
Page %pagenumber% of %pagecount%
```

However, if I use both background pages and `%pagecount%` the background page gets counted in the number of pages. This is expected because a background page is no different from a regular page.

I don't see how this behavior could easily be changed with the current data schema within drawio. Does anyone have any ideas about ways that I could cheat? Like defining some sort of custom property that has the value `%pagecount% - 1`? It's okay if it's a little hackey.

**Comment by jwilliams-shadowsoft** [comment]
It seems to me that being a background page is an attribute of the page, and that we should be able to mark a page as a background. Then, the logic for %pagecount% could be altered to ignore those background pages.

**Comment by cameronweston** [comment]
I'm using drawio in the exact same way and would love the option to exclude the background page in the pagecount.

**Comment by MrDanielGerber** [comment]
I agree! That would be a significant QoL-feature that would allow me to create more official engineering documentation in draw.io

**Comment by JMalton** [comment]
It would also be appropriate that, if a sheet is locked as "background", it doesn't get exported.  Currently, I need to hard-code the number of pages because the background page gets included in the %pagecount% placeholder value, then export all the pages except the background one.  Unfortunately, (with PDF export anyway), that ends up exporting the required pages, plus a repeat of several pages on the end.

**Comment by alderg** [comment]
You can use `%pagecount-1%` to leave a background page out of the count, e.g. `Page %pagenumber% of %pagecount-1%` (and `%pagenumber-1%` if the background page comes first). To leave it out of a PDF, select the page range without it in File > Export as > PDF.
